addons/mrp/models/product.py

```diff
@@ -195,3 +195,18 @@ def action_open_quants(self):
             res['context']['single_product'] = False
             res['context'].pop('default_product_tmpl_id', None)
         return res
+
+    def _search_qty_available_new(self, operator, value, lot_id=False, owner_id=False, package_id=False):
+        '''extending the method in stock.product to take into account kits'''
+        product_ids = set(super(ProductProduct, self)._search_qty_available_new(operator, value, lot_id, owner_id, package_id))
+        boms = self.env['mrp.bom'].read_group([('type', "=", 'phantom')], ['product_tmpl_id', 'ids:array_agg(id)'], ['product_tmpl_id', 'id'])
+        for kit in boms:
+            product_id = self.env['product.template'].browse(kit['product_tmpl_id'][0]).product_variant_ids.id
+            bom_lines = self.env['mrp.bom.line'].read_group([('bom_id', "=", kit['ids'][0])], ['product_id'], ['product_id'])
```

Comment:
this could be done without read group:

```suggestion
            components = self.env['mrp.bom'].browse(kit['ids'][0]).bom_line_ids.product_id
            if set(components.ids) <= product_ids:
                product_ids.add(product_id)
```